src/common.ts

```diff
@@ -67,11 +67,7 @@ export class ContractService<T extends ChainId, E = {}> extends Service<E> {
     super(chainId, ctx);
     this.address = address;
 
-    this.contract = new WrappedContract(
-      this.address,
-      // @ts-ignore
-      this.constructor.abi,
-      ctx
-    );
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Unsure how to type this, the compiler was really really unhappy with the use of `@ts-ignore`, so I think we should try and avoid that at all costs. Casted to `any`, but it would be nice to be better typed if anyone knows what type that should be.

Reply:
good question, not sure what the type should be

Reply:
not sure about the intention of `abi` being static, so not sure. Seems ABIs are typed as ContractInterface, not sure here why `static abi: string[] = [];`. Its ok to leave it as any, and in another PR we can give it a deeper thought.